stripe/src/main/java/com/stripe/android/model/Stripe3ds2Fingerprint.java

```diff
@@ -43,30 +49,39 @@ private Stripe3ds2Fingerprint(@NonNull String source,
         this.directoryServerEncryption = directoryServerEncryption;
     }
 
-    public static class DirectoryServerEncryption {
+    public static final class DirectoryServerEncryption {
         private static final String FIELD_DIRECTORY_SERVER_ID = "directory_server_id";
         private static final String FIELD_CERTIFICATE = "certificate";
         private static final String FIELD_KEY_ID = "key_id";
 
         @NonNull public final String directoryServerId;
-        @NonNull public final String certificate;
+        @NonNull public final Certificate certificate;
         @Nullable public final String keyId;
 
-        private DirectoryServerEncryption(@NonNull String directoryServerId,
-                                          @NonNull String certificate,
-                                          @Nullable String keyId) {
+        @VisibleForTesting
+        DirectoryServerEncryption(@NonNull String directoryServerId,
+                                  @NonNull String certificate,
```

Comment:
nit: maybe call this certificateContents or certificateString to disambiguate from a real certificate